yoyodyne/datasets.py

```diff
@@ -532,26 +518,49 @@ def __getitem__(
         """
         source, features, target = self.samples[idx]
         source_encoded = self.encode(self.source_symbol2i, source)
-        features_encoded = self.encode(
-            self.source_symbol2i,
-            features,
-            add_start_tag=False,
-            add_end_tag=False,
-        )
+        features_encoded = self.encode_features(features)
         target_encoded = (
             self.encode(self.target_symbol2i, target, add_start_tag=False)
             if self.target_col
             else None
         )
         return source_encoded, features_encoded, target_encoded
 
+    def encode_features(
+        self,
+        features: List[str],
+    ) -> torch.Tensor:
+        """Encodes a sequence as a tensor of indices with word boundary IDs.
+
+        This essentially copies behavior of encode but limits return values
+        to only features seen from initialization, so unknown feature
+        values are not permitted.
+
+        Args:
+            features (List[str]): features to be encoded.
+
+        Returns:
+            torch.Tensor: the encoded tensor.
+        """
+        sequence = []
+        for feature in features:
+            if feature in self.source_symbol2i:
+                sequence.append(self.source_symbol2i[feature])
+            else:
+                raise Error(
+                    f"Feature {feature!r} seen during inference was not "
```

Comment:
This ignores cases of unknown features. Did we revert the decision from the previous discussion?

Reply:
Merge error, fixed.